ui/raidboss/emulator/overrides/RaidEmulatorOverlayApiHook.js

```diff
@@ -1,16 +1,20 @@
+import { callOverlayHandler, addOverlayListener, removeOverlayListener, setOverride } from '../../../../resources/overlay_plugin_api';
+
 export default class RaidEmulatorOverlayApiHook {
   constructor(emulator) {
     this.emulator = emulator;
     this.originalDispatch = window.dispatchOverlayEvent;
-    this.originalAdd = window.addOverlayListener;
-    this.originalRemove = window.removeOverlayListener;
-    this.originalCall = window.callOverlayHandler;
+    this.originalAdd = addOverlayListener;
+    this.originalRemove = removeOverlayListener;
+    this.originalCall = callOverlayHandler;
     this.timestampOffset = 0;
 
-    window.dispatchOverlayEvent = this.dispatch.bind(this);
-    window.addOverlayListener = this.add.bind(this);
-    window.removeOverlayListener = this.remove.bind(this);
-    window.callOverlayHandler = this.call.bind(this);
+    setOverride({
+      addOverlayListenerOverride: this.add.bind(this),
+      removeOverlayListenerOverride: this.remove.bind(this),
+      callOverlayHandlerOverride: this.call.bind(this),
+      dispatchOverlayEventOverride: this.add.bind(this),
+    });
```

Comment:
I feel like there's a better way to handle this given that the code is being refactored anyways, but I can't really think of exactly how this should be handled.

Functionally this code exists only to override the `getCombatants` handler, so I wonder if it would make sense instead to be able to add an override per-handler, something like:

```js
setCallOverride('getCombatants', (msg, originalCall) => {/* some func */});
```

Thoughts/ideas?

Reply:
I think this is probably ok for now, and maybe can follow-up with cleanup if you'd like.  Was there a reason that the raid emulator was overriding the other things if it doesn't need to?

Reply:
Yeah, it's fine if this is cleaned up in a followup, as long as nothing else is using `setOverride` here other than raidemulator.

When I was writing that code my initial goal was to see if I could completely remove the need for ACT running/having a WS to connect to, so I ended up intercepting everything and just never removed the unneeded intercepts after that initial goal didn't pan out.

Reply:
I don't know much about this Emulator.
I want to make sure if I need to change this part of the code before merging?

Reply:
You don't need to change anything here.

I'll follow up with a PR when I get the time to handle this in a more clean manner. It might be when I convert raidemulator to typescript.